src/prefect/new_task_engine.py

```diff
@@ -262,10 +262,26 @@ async def create_task_run(self, client: PrefectClient) -> TaskRun:
             k: await collect_task_run_inputs(v) for k, v in self.parameters.items()
         }
 
-        # anticipate nested runs
+        # check if this task has a parent task run based on running in another
+        # task run's existing context. A task run is only considered a parent if
+        # it is in the same flow run (because otherwise presumably the child is
+        # in a subflow, so the subflow serves as the parent) or if there is no
+        # flow run
         task_run_ctx = TaskRunContext.get()
         if task_run_ctx:
-            task_inputs["wait_for"] = [TaskRunResult(id=task_run_ctx.task_run.id)]
+            # there is no flow run
+            if not flow_run_ctx:
+                task_inputs["__parents__"] = [
```

Comment:
I noticed the `__parents__` key is a list. Is there a scenario where a task could have multiple parents?

Reply:
Ha, @cicdw and I debated exactly this. We really wanted to say no but here is what we learned:
- the Server API already types this as a list, so the point is unfortunately moot and it's going to have to be a list even if just for compatibility reasons
- it turns out... yes there is a situation! In #13138 I experimented with extending the engine to support generator tasks. The jury is still out on whether we want to support that (it's cool, but creates some question marks for how things like task.submit work) but if a task received inputs from two different running generators, then one of the rules we identified for that PR is that the task would actually consider both generators to be its parents! So again one example does not a rule make, but I think it's not out of the question that we'll find situations where a task has multiple parent tasks as well as multiple upstream dependencies.